### PR TITLE
Add Analytics for Themes

### PR DESCRIPTION
**Pull request by kwonye**
Addresses #3376 (Also sneaks in analytics for search)

Would be great to have in before 4.8 branch off (Nov 16, 2015)

Needs review: @daniloercoli

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java [hunk 1]
```diff
@@ -352,8 +352,9 @@ private void activateTheme(String themeId) {
             @Override
             public void onResponse(JSONObject response) {
                 WordPress.wpDB.setCurrentTheme(siteId, newThemeId);
+                Theme newTheme = WordPress.wpDB.getTheme(siteId, newThemeId);
+                AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.THEMES_CHANGED_THEME);
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java]: Can we add the `id` of the theme the user has switched to as property of the event? 
Can we make sure the `blog_id` is added to properties as well?

File: WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java [hunk 2]
```diff
@@ -392,6 +393,20 @@ private void startWebActivity(String themeId, ThemeWebActivity.ThemeWebActivityT
...
+                    case DEMO:
+                        AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.THEMES_DEMO_ACCESSED);
+                        break;
+                    case DETAILS:
+                        AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.THEMES_DETAILS_ACCESSED);
+                        break;
+                    case SUPPORT:
+                        AnalyticsUtils.trackWithCurrentBlogDetails(AnalyticsTracker.Stat.THEMES_SUPPORT_ACCESSED);
+                        break;
+                }
```
Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/ui/themes/ThemeBrowserActivity.java]: As above, for all of these calls can we please add the `blog_id` and the `theme_id` to properties?

See https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java for details. 

note: Make sure that the "remote blog id" is correctly used it. `String blogID = StatsUtils.getBlogId(blog);` See the class above for details ;)

**Comment by kwonye** [comment]
@daniloercoli both "blog_id" and "theme_id" is added. However, adding "blog_id" is slightly confusing to me. If I call `AnalyticsUtils.trackWithCurrentBlogDetails(stat, properties)` is that not already included in the properties list?

I split up the PR into two commits, so if `blog_id` is not necessary, please reverse commit 62d4fbd for me. Also, just to double-make-sure, can you please merge once you give the :shipit: please? Thanks :blush:

**Comment by daniloercoli** [comment]
Sorry @kwonye, you're right. The calls with `AnalyticsUtils.trackWithCurrentBlogDetails(stat, properties)` already contain the `blog_id`. I removed that commit.

**Comment by daniloercoli** [comment]
:shipit:

**Pull request merged**